projects/addon-doc/src/components/code/parse-code-block.ts

```diff
@@ -0,0 +1,19 @@
+import MarkdownIt from 'markdown-it';
+import Token from 'markdown-it/lib/token';
+
+export function tryParseMarkdownCodeBlock(text: string): string[] {
+    const result: string[] = [];
+    const tokens: Token[] = new MarkdownIt().parse(text, {});
+
+    for (const token of tokens) {
+        if (token.tag === 'code') {
+            result.push(token.content);
+        }
+    }
+
+    if (result.length === 0) {
+        result.push(text);
+    }
+
+    return result;
```

Comment:
This seems a bit off in terms of the rest of our codestyle. Let's rewrite it this way:

```ts
const tokens = new MarkdownIt().parse(text, {});
const result = tokens.filter(({tag}) => tag === 'code').map(({content}) => content);

return result.length ? result : [text];
```

Reply:
fixed